ddclient

```diff
@@ -2307,6 +2306,25 @@ sub get_ip {
     return $ip;
 }
 
+######################################################################
+## is_ipv4() validates if string is valid IPv4 address and only
+## a valid IPv4 address (no preceding or trailing spaces/characters)
+######################################################################
+sub is_ipv4 {
+    my ($value) = @_;
+    return (length($value // '') != 0) && ($value eq ipv4_match($value));
```

Comment:
`ipv4_match` returns `undef` if there is no match. If that return value is directly compared with `$value` we'll get a warning.

```suggestion
    return (length($value // '') != 0) && ($value eq (ipv4_match($value) // ''));
```